packages/angular/build/src/tools/vite/id-prefix-plugin.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import type { Plugin } from 'vite';
+
+// NOTE: the implementation for this Vite plugin is roughly based on:
+// https://github.com/MilanKovacic/vite-plugin-externalize-dependencies
+
+const VITE_ID_PREFIX = '/@id/';
+const vitePrefixRegex = new RegExp(VITE_ID_PREFIX, 'g');
+
+const escapeRegexSpecialChars = (inputString: string): string => {
+  return inputString.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+};
+
+export const createRemoveIdPrefixPlugin = (externals: string[]): Plugin => ({
+  name: 'vite-plugin-remove-id-prefix',
+  apply: 'serve',
+  configResolved: (resolvedConfig) => {
+    // don't do anything when the list of externals is empty
+    if (externals.length === 0) {
+      return;
+    }
+
+    const escapedExternals = externals.map(escapeRegexSpecialChars);
+    const prefixedExternalRegex = new RegExp(
+      `${VITE_ID_PREFIX}(${[...escapedExternals].join('|')})`,
```

Comment:
Since `escapedExternals` is an array, there shouldn't be a need to copy it here.

Reply:
Good catch. Updated.